src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/SessionBlittableJsonConverter.cs

```diff
@@ -71,9 +72,44 @@ public object FromBlittable(Type type, ref BlittableJsonReaderObject json, strin
             }
             catch (Exception ex)
             {
-                throw new InvalidOperationException($"Could not convert document {id} to entity of type {type}",
+                string jsonAsString = TryReadBlittableAsString(json);
+
+                throw new InvalidOperationException($"Could not convert document {id} to entity of type {type}. Json: {jsonAsString}",
                     ex);
             }
+
+            string TryReadBlittableAsString(BlittableJsonReaderObject jsonToRead)
+            {
+                var jsString = string.Empty;
+
+                using (var memoryStream = new MemoryStream())
+                {
+                    try
+                    {
+                        jsonToRead.WriteJsonToAsync(memoryStream).AsTask().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
why async here?

Reply:
There is no sync method there since we moved to async - https://github.com/ravendb/ravendb/pull/11321